src/core/lib/gpr/alloc.cc

```diff
@@ -47,6 +83,15 @@ void gpr_set_allocation_functions(gpr_allocation_functions functions) {
   if (functions.zalloc_fn == nullptr) {
     functions.zalloc_fn = zalloc_with_gpr_malloc;
   }
+  GPR_ASSERT((functions.aligned_alloc_fn == nullptr) ==
+             (functions.aligned_free_fn == nullptr));
+  GPR_ASSERT((functions.cacheline_alloc_fn == nullptr) ==
+             (functions.aligned_free_fn == nullptr));
+  if (functions.aligned_alloc_fn == nullptr) {
+    functions.aligned_alloc_fn = aligned_alloc_with_gpr_malloc;
```

Comment:
This should be `gpr_malloc_aligned`?
Same for the rest of them.

Reply:
This version of the PR is different than what I have seen initially. I think we should use `platform_malloc_aligned` here.

My point is that `platform_malloc_aligned` is the default and `aligned_alloc_with_gpr_malloc` is a fallback that it uses. Otherwise, you'll make old binaries that are not aware of your nice change, just slower than they should be.

Reply:
Done.